src/main/java/duke/pack/Parser.java

```diff
@@ -0,0 +1,146 @@
+package duke.pack;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * represents a parser to interpret the user's commands
+ */
+public class Parser {
+    public Parser() {
+    }
+
+    /**
+     * Interprets the user's command
+     * @param command input of the user
+     * @return a Command that indicates what the user wants
+     * @throws DukeException if command is not correctly inputted
+     */
+    public static Command parseCommand(String command) throws DukeException {
+        String[] comm = command.split(" ");
+
+        if (command.equals("list")) {
+            // list
+            Command c = new ListCommand();
+            return c;
+
+        } else if (comm[0].equals("find")) {
+            // find
+            if (comm.length == 1) {
+                throw new DukeException("    Oh no! You have to specify what you want to find!");
+            }
+
+            Command c = new FindCommand(comm[1]);
+            return c;
+
+        } else if (comm[0].equals("done")) {
+            // done
+            if (comm.length == 1) {
+                throw new DukeException("    Oh no! You have to specify which task is done!");
+            }
+
+            Command c = new DoneCommand(Integer.parseInt(comm[1]));
+            return c;
+
+        } else if (comm[0].equals("delete")) {
+            // delete
+            if (comm.length == 1) {
+                throw new DukeException("    Oh no! You have to specify which task to delete!");
+            }
+
+            Command c = new DeleteCommand(Integer.parseInt(comm[1]));
+            return c;
+
+        } else if (comm[0].equals("todo")) {
+            // to-do
+            // if no description is given
+            if (comm.length == 1) {
+                throw new DukeException("    Oh no! A todo cannot be empty!");
+            }
+
+            String[] arr = command.split("todo");
+            Task t = new Todo(arr[1].trim(), arr[1]);
+
+            Command c = new AddCommand(t);
+            return c;
+
+        } else if (comm[0].equals("event")) {
+            // event
+            // if no description is given
+            if (comm.length == 1) {
+                throw new DukeException("    Oh no! An event cannot be empty!");
```

Comment:
The parseCommand method is really lengthy which will reduce the readability of the code. One suggestion would be to extract outset of related code segments as methods. E.g. Extract event command-specific validations Or you can have different parsers as well.